modules/Sources/Features/NHSendFlow/Recipient/RecipientView.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  RecipientView.swift
+//  
+//
+//  Created by Matthew Watt on 7/23/23.
+//
+
+import ComposableArchitecture
+import Generated
+import SwiftUI
+import UIComponents
+
+public struct RecipientView: View {
+    let store: RecipientStore
+    
+    public init(store: RecipientStore) {
+        self.store = store
+    }
+    
+    @FocusState private var isRecipientEditorFocused: Bool
+    
+    public var body: some View {
+        WithViewStore(store) { viewStore in
+            VStack {
+                NighthawkHeading(
+                    title: L10n.Nighthawk.TransferTab.Recipient.chooseRecipient
+                )
+                .padding(.bottom, 40)
+                .onTapGesture {
+                    isRecipientEditorFocused = false
+                }
+                
+                NHTextField(
+                    placeholder: L10n.Nighthawk.TransferTab.Recipient.addZAddress,
+                    text: viewStore.bindingForRedactableRecipient(viewStore.recipient),
+                    foregroundColor: Asset.Colors.Nighthawk.parmaviolet.color,
+                    inputAccessoryView: {
+                        addressFieldAccessoryView(with: viewStore)
+                    }
+                )
+                .padding(.horizontal, 24)
+                .foregroundColor(.green)
+                .focused($isRecipientEditorFocused)
+                
+                if viewStore.canPasteAddress {
+                    Button(
+                        L10n.Nighthawk.TransferTab.Recipient.pasteFromClipboard,
+                        action: { viewStore.send(.pasteFromClipboardTapped) }
+                    )
+                    .buttonStyle(.nighthawkDashed())
+                }
+                
+                Spacer()
+                
+                Button(
+                    L10n.Nighthawk.TransferTab.Recipient.continue,
+                    action: { viewStore.send(.continueTapped) }
+                )
+                .buttonStyle(.nighthawkPrimary())
+                .disabled(!viewStore.isRecipientValid)
```

Comment:
[maybe-blocking]

disabling the button might not alter its style so the button would look the same for the user but it won't react to touches.

Reply:
`NighthawkPrimaryButtonStyle` and all of its siblings utilize the following:
`@Environment(\.isEnabled) private var isEnabled: Bool`

The style adapts to its environment, which can be controlled via the `.disabled()` modifier.